src/easyscience/fitting/minimizers/minimizer_lmfit.py

```diff
@@ -257,6 +177,56 @@ def convert_to_par_object(parameter: Parameter) -> LMParameter:
             brute_step=None,
         )
 
+    def _make_model(self, pars: Optional[LMParameters] = None) -> LMModel:
+        """
+        Generate a lmfit model from the supplied `fit_function` and parameters in the base object.
+
+        :return: Callable lmfit model
+        :rtype: LMModel
+        """
+        # Generate the fitting function
+        fit_func = self._generate_fit_function()
+
+        self._fit_function = fit_func
+
+        if pars is None:
+            pars = self._cached_pars
+        # Create the model
+        model = LMModel(
+            fit_func,
+            independent_vars=['x'],
+            param_names=[MINIMIZER_PARAMETER_PREFIX + str(key) for key in pars.keys()],
+        )
+        # Assign values from the `Parameter` to the model
+        for name, item in pars.items():
+            if isinstance(item, LMParameter):
+                value = item.value
+            else:
+                ## TODO clean when full move to new_variable
+                from easyscience.Objects.new_variable import Parameter
+
+                if isinstance(item, Parameter):
+                    value = item.value
+                else:
+                    value = item.raw_value
+
+            model.set_param_hint(MINIMIZER_PARAMETER_PREFIX + str(name), value=value, min=item.min, max=item.max)
+
+        # Cache the model for later reference
+        self._cached_model = model
+        return model
+
+    def _generate_fit_function(self) -> Callable:
+        """
+        Using the user supplied `fit_function`, wrap it in such a way we can update `Parameter` on
+        iterations.
+
+        :return: a fit function which is compatible with lmfit models
+        """
+        fit_function = super()._generate_fit_function()
+        fit_function.__signature__ = self._create_signature(self._cached_pars)
+        return fit_function
```

Comment:
`_generate_fit_function` here and in bumps have the same code. They call base class implementation then add signature based on `_cached_pars` which is also accessible in the base class. Should these be just merged with the base class method?

Reply:
You are right.  Moved the signature generation to the base class and adjusted tests